cfht2caom2/cfht_name.py

```diff
@@ -424,3 +431,546 @@ def remove_extensions(name):
             .replace('.gz', '')
             .replace('.hdf5', '')
         )
+
+
+def get_instrument(headers, entry):
+    """
+    SF - 15-04-20 - slack - what if there's no INSTRUME or DETECTOR
+    keyword?
+    something like: if CFHT: if has(INSTRUME) then
+    MEGA else if NEXTEND>30 then MEGA else...
+
+    On MegaPrime vs MegaCam values from SVO:
+    megaprime == full instrument (optics+camera) ,
+    megacam == camera
+
+    SVO has mistakes (they say megacam=CCD+mirror+optics), but i think the
+    Megaprime entry has a lot more filters. So to be consistent, we should
+    always use the Megaprime entry
+
+    SGo - make the 'always use Megaprime' happen here by setting the
+    instrument to MegaPrime.
+
+    :param headers: astropy fits headers
+    :param entry: string for error logging
+    :return: md.Inst instance
+    """
+    if StorageName.is_hdf5(entry):
+        inst = Inst.SITELLE
+    else:
+        nextend = None
+        detector = None
+        instrument = headers[0].get('INSTRUME')
+        if instrument is None:
+            if len(headers) > 1:
+                instrument = headers[1].get('INSTRUME')
+            if instrument is None:
+                instrument = headers[0].get('DETECTOR')
+                if instrument is None:
+                    if len(headers) > 1:
+                        instrument = headers[1].get('DETECTOR')
```

Comment:
Wondering if this works (haven't tested it)

```suggestion
        instrument = headers[0].get('INSTRUME') or (len(headers) > 1 and headers[1].get('INSTRUME')) or headers[0].get('DETECTOR') or (len(headers) > 1 and headers[1].get('DETECTOR'))
```

Reply:
I replaced this with a function `get_keyword` that does the [0, 1] lookup for any keyword.